### PR TITLE
creating model class, repository, service class and mapper for subject

### PR DESCRIPTION
**Pull request by farhadnajafov**

### CODE REVIEW

**Review by aytiqaqash — CHANGES_REQUESTED** [review]
We are using camel cases in Model classes based on the name convention and suggesting changing "course_id" and "language_id" to "courseId" and "languageId" accordingly. Could you please change the Mapper file as well?

**Comment by farhadnajafov** [comment]
> We are using camel cases in Model classes based on the name convention and suggesting changing "course_id" and "language_id" to "courseId" and "languageId" accordingly. Could you please change the Mapper file as well?

@aytiqaqash I changed. Please review agin

**Review by aytiqaqash — APPROVED** [review]
LGTM

File: src/main/java/com/webperside/courseerpbackend/services/subject/SubjectService.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.List;
+
+public interface SubjectService {
+
+    List<Subject> findAll();
+    Subject findById(Long id);
+    void  insert(Subject subject);
+    void update (@RequestBody Subject subject);
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/services/subject/SubjectService.java:13]: Remove @requestbody from the repository's method

File: src/main/java/com/webperside/courseerpbackend/repository/SubjectRepository.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@Mapper
+public interface SubjectRepository {
+
+    List<Subject> findAll();
+    Optional<Subject> findById(@Param("id") Long id);
+    void insert(Subject subject);
+    void update (@RequestBody Subject subject);
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/repository/SubjectRepository.java:17]: Remove @requestbody from the repository's method

**Review by hamidsultanzadeh — CHANGES_REQUESTED** [review]

**Review by aytiqaqash — COMMENTED** [review]
LGTM

**Pull request merged**